index.js

```diff
@@ -0,0 +1,27 @@
+//function validatePassword
+const validatePassword = (password) => {
+
+  if (password.length < 8) {
+    return false
+  }
+
+  if (!(/[a-z]/.test(password))) {
+    return false
+  }
+
+  if (!(/[A-Z]/.test(password))) {
+    return false
+  }
+
+  if (!(/[0-9]/.test(password))) {
+    return false
+  }
+
+  if (!(/[$&+,:;=?@#|'<>.^*()%!-]/.test(password))) {
+    return false
+  }
+
+  return true
```

Comment:
These could be broken into smaller functions that would help with readability and maintainability. Even if not you could combine the conditions and not need all the if statements.

```JavaScript
const validatePassword = (password) => {
  return password.length >= 8 &&
    /[a-z]/.test(password) &&
    /[A-Z]/.test(password) &&
    /[0-9]/.test(password) &&
    /[$&+,:;=?@#|'<>.^*()%!-]/.test(password)
}
```

Which looks better when they are sub functions.

```JavaScript
const isValidLength = (password) => password.length >= 8

const hasLowerCase = (password) => /[a-z]/.test(password)

const hasUpperCase = (password) => /[A-Z]/.test(password)

const hasNumeric = (password) => /[0-9]/.test(password)

const hasSpecialCharacter = (password) => /[$&+,:;=?@#|'<>.^*()%!-]/.test(password)

const validatePassword = (password) => {
  return isValidLength(password) &&
    hasLowerCase(password) &&
    hasUpperCase(password) &&
    hasNumeric(password) &&
    hasSpecialCharacter(password)
}
```